rdsbroker/broker.go

```diff
@@ -387,20 +391,79 @@ func (b *RDSBroker) LastOperation(instanceID string) (brokerapi.LastOperationRes
 	return lastOperationResponse, nil
 }
 
+func (b *RDSBroker) CheckAndRotateCredentials() error {
+	dbInstanceDetailsList, err := b.dbInstance.DescribeByTag("Broker Name", b.brokerName)
+	if err != nil {
+		b.logger.Error("Could not retrieve the list of DB instances", err)
+		return err
+	}
+
+	for _, dbInstanceDetails := range dbInstanceDetailsList {
+		dbAddress := dbInstanceDetails.Address
+		dbPort := dbInstanceDetails.Port
+		masterUsername := dbInstanceDetails.MasterUsername
+		serviceInstanceID := b.dbInstanceIdentifierToServiceInstanceID(dbInstanceDetails.Identifier)
+
+		var dbName string
+		if dbInstanceDetails.DBName != "" {
+			dbName = dbInstanceDetails.DBName
+		} else {
+			dbName = b.dbName(dbInstanceDetails.Identifier)
+		}
+
+		sqlEngine, err := b.sqlProvider.GetSQLEngine(dbInstanceDetails.Engine)
+		if err != nil {
+			b.logger.Error("Could not get the SQL Engine", err)
+			return err
+		}
+
+		err = sqlEngine.Open(dbAddress, dbPort, dbName, masterUsername, b.masterPassword(serviceInstanceID))
+		if err != nil {
+			b.logger.Error("Error opening connection to the database", err)
+			return err
```

Comment:
I'm unsure about returning from this loop if we get an error on a single instance. It would mean that if all passwords need rotating but one instance is down then some instances could be left with the incorrect password.

To change that though we'd need to either return `[]error` or split the code within the loop out to a separate function that we call/log multiple times.

What do you think?